src/sinol_make/oiejq/__init__.py

```diff
@@ -0,0 +1,124 @@
+
+    try:
+        request = requests.get('https://oij.edu.pl/zawodnik/srodowisko/oiejq.tar.gz')
+    except requests.exceptions.ConnectionError:
+        raise Exception('Couldn\'t download oiejq (https://oij.edu.pl/zawodnik/srodowisko/oiejq.tar.gz couldn\'t connect)')
+    if request.status_code != 200:
+        raise Exception('Couldn\'t download oiejq (https://oij.edu.pl/zawodnik/srodowisko/oiejq.tar.gz returned status code: ' + str(request.status_code) + ')')
+
+    # oiejq is downloaded to a temporary directory and not to the `cache` dir,
+    # as there is no guarantee that the current directory is the package directory.
+    # The `cache` dir is only used for files that are part of the package and those
+    # that the package creator might want to look into.
+    with tempfile.TemporaryDirectory() as tmpdir:
+        oiejq_path = os.path.join(tmpdir, 'oiejq.tar.gz')
+        with open(oiejq_path, 'wb') as oiejq_file:
+            oiejq_file.write(request.content)
+
+        def strip(tar):
+            l = len('oiejq/')
+            for member in tar.getmembers():
+                member.name = member.name[l:]
+                yield member
+
+        with tarfile.open(oiejq_path) as tar:
+            tar.extractall(path=os.path.expanduser('~/.local/bin'), members=strip(tar))
+        os.rename(os.path.expanduser('~/.local/bin/oiejq.sh'), os.path.expanduser('~/.local/bin/oiejq'))
+
+    return check_oiejq()
+
+
+def get_oiejq_path():
+    if not check_oiejq():
+        return None
+
+    def check(path):
+        p = subprocess.Popen([path], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        p.wait()
+        p.kill()
+        if p.returncode == 0:
+            return True
+        else:
+            return False
+
+    if check(os.path.expanduser('~/.local/bin/oiejq')):
+        return os.path.expanduser('~/.local/bin/oiejq')
+    else:
+        return None
+
+
+def check_perf_counters_enabled():
+    """
+    Checks if `kernel.perf_event_paranoid` is set to -1.
+    :return:
+    """
+    if sys.platform != 'linux' or not check_oiejq():
+        return
+
+    oiejq = get_oiejq_path()
+    test_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'perf_test.py')
+    python_executable = sys.executable
+
+    # subprocess.Pipe is not used, because than the code would hang on process.communicate()
+    with tempfile.TemporaryFile() as tmpfile:
+        process = subprocess.Popen([oiejq, python_executable, test_file], stdout=tmpfile, stderr=subprocess.DEVNULL)
+        process.wait()
+        tmpfile.seek(0)
+        output = tmpfile.read().decode('utf-8')
+        process.terminate()
+
+    if output != "Test string\n":
+        util.exit_with_error("You don't have permission to use perf counters required by `oiejq`. "
+                             "Please run `sudo sysctl kernel.perf_event_paranoid=-1` or "
+                             "add `kernel.perf_event_paranoid=-1` to `/etc/sysctl.conf` and reboot.")
```

Comment:
The "sysctl" command is *not* equivalent to the "add to /etc/sysctl.conf", right? The first one works only for the current shell session and the second one is permament? (I'm not sure).

Reply:
The first works until the system is rebooted

Reply:
```suggestion
        util.exit_with_error("You don't have permission to use perf counters required by `oiejq`. "
                             "Please run `sudo sysctl kernel.perf_event_paranoid=-1` or "
                             "add `kernel.perf_event_paranoid=-1` to `/etc/sysctl.conf` and reboot.")
```
I would prefer something that explains the situation a bit more:

```
To use the recommended tool for measuring time called oiejq, please:
- execute `sudo sysctl kernel.perf_event_paranoid=-1` to make oiejq work for the current system session,
- or add `kernel.perf_event_paranoid=-1` to `/etc/sysctl.conf` and reboot to permamently make oiejq work.
For more details, see [link or path to readme of oiejq that explains those commands].
```

Let's also manually split the longer printed messages, so that they fit in terminals of users with 80 character width.

As oiejq doesn't work on MacOS and Windows, what do you think of writing an error that the recommended tool for measuring time oiejq doesn't work on those systems and the user can specify another tool for measuring time through the `--time` command?

Reply:
I changed the message, but in oiejq's and sio2jail's readme it isn't explained what this command does.

The message for `oiejq` on Windows and macOS is here:
https://github.com/sio2project/sinol-make/blob/aea45b8d275712178fd5de86695551f6132dd55d/src/sinol_make/commands/run/__init__.py#L919-L921
but I think that it should be change to something like:
```
`oiejq` is only available on Linux. You can specify another time tool with `--time` flag.
```

Reply:
Let's also explain that `oiejq` is the recommended way of measuring time. Maybe we could link the README or copy-paste parts of it to this error message.

Reply:
I changed this message